frontend/src/pages/CompareV2.tsx

```diff
@@ -15,9 +15,95 @@
  */
 
 import React from 'react';
+import { useQuery } from 'react-query';
+import { ApiRunDetail } from 'src/apis/run';
+import { QUERY_PARAMS } from 'src/components/Router';
 import { commonCss } from 'src/Css';
+import { Apis } from 'src/lib/Apis';
+import { URLParser } from 'src/lib/URLParser';
+import { errorToMessage } from 'src/lib/Utils';
+import {
+  getExecutionsFromContext,
+  getKfpV2RunContext,
+  getOutputLinkedArtifactsInExecution,
+  LinkedArtifact,
+} from 'src/mlmd/MlmdUtils';
+import { Execution } from 'src/third_party/mlmd';
+import { PageProps } from './Page';
+
+interface ExecutionArtifacts {
+  execution: Execution;
+  linkedArtifacts: LinkedArtifact[];
+}
+
+interface RunArtifacts {
+  run: ApiRunDetail;
+  executionArtifacts: ExecutionArtifacts[];
+}
```

Comment:
Note: These objects were designed so that I can easily loop through all of the artifacts within each execution on each run, which is how the information will be sorted and displayed.